dask_saturn/core.py

```diff
@@ -50,6 +52,16 @@ def scheduler_address(self):
     def dashboard_link(self):
         return self._dashboard_link
 
+    def __await__(self):
+        async def _():
+            async with self.lock:
+                if self.status == "created":
+                    await self.start()
+                    assert self.status == "running"
+            return self
+
+        return _().__await__()
```

Comment:
This looks identical to the existing `__await__()` on `SpecCluster` (https://github.com/dask/distributed/blob/f748d7d539df0285b46b3c966c587dfd76bf3e48/distributed/deploy/spec.py#L46). Aren't we already getting it for free?

Reply:
The `__await__` that is defined on `SpecCluster` is actually further down (https://github.com/dask/distributed/blob/15591929f3a6b7b390a7a5394e1f53fe6a6c16f4/distributed/deploy/spec.py#L361-L371) and includes calls to `self.scheduler` and `self.workers` both of which we are not implementing

Reply:
No worries, I was like wait is that what I did?!